### PR TITLE
Move model structure params into a file.

### PR DESCRIPTION
**Pull request by ntadimeti**
Added option to parse arguments from config file as well as command line. Future flow will be : 
1. Current model structure details are moved into model_structure.yaml file. 
2. One can still specify the command line argument to override the yaml file configurations
3. Hierarchy : Command line > config file > default config

config features : 
-> default model_structure.yaml config file saved under Atacworks/configs
-> one can provide a custom config file using --config_mparams flag
-> alternatively, one can also skip the config file altogether by providing all the parameters on command line

Other features : 
-> remove the "train" folder under claragenomics/dl4atac
-> create a models folder
    -> added a model_utils.py script to parse model params and build models.
-> added a backward compatibility test script under tests/run_v0.1.sh to make sure we can run the scripts without any configs as well.

### CODE REVIEW

File: claragenomics/dl4atac/utils.py [hunk 1]
```diff
@@ -51,6 +51,25 @@ def make_experiment_dir(label, out_home, timestamp=True, create_latest_symlink=T
...
+    """
+    myprint("Saving config files to {}...".format(exp_dir))
+    # Save custom config file if provided
+    if config_file_path:
+        print ("comes here")
+        shutil.copy(config_file_path, exp_dir)
+    # Save default config
+    config_path = os.path.join(root_dir, "configs")
+    dst_path = os.path.join(exp_dir, "configs")
+    shutil.copytree(config_path, dst_path)
```
Comment (tijyojwad) [inline claragenomics/dl4atac/utils.py]: will this copy the whole config folder? Would be better to copy only those config files used for the run

File: claragenomics/dl4atac/utils.py [hunk 2]
```diff
@@ -51,6 +51,25 @@ def make_experiment_dir(label, out_home, timestamp=True, create_latest_symlink=T
...
+    Args:
+        exp_dir : Directory where the models are being saved.
+        root_dir : Directory where the main script resides. It's assumed that the configs
+                   directory is at the same level as the script.
+        config_file_path : Path to a custom config file if any.
+    """
+    myprint("Saving config files to {}...".format(exp_dir))
+    # Save custom config file if provided
+    if config_file_path:
+        print ("comes here")
```
Comment (tijyojwad) [inline claragenomics/dl4atac/utils.py]: debug line can be removed

File: claragenomics/dl4atac/models/model_utils.py [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+_logger.setLevel(logging.INFO)
+_logger.addHandler(_handler)
+
+
+def model_args_v1():
+    parser = configargparse.ArgParser(default_config_files=['configs/model_structure.yaml'])
+    parser.add('--model_config', required=False, is_config_file=True, help='config file path')
+
+    # Model architecture args
+    parser.add('--model', type=str, help='model type', choices=(
```
Comment (tijyojwad) [inline claragenomics/dl4atac/models/model_utils.py]: can you make all these options required so they either have to be present in one of the config files or on the command line?

File: model_args.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
+#
+# Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+#
+# NVIDIA CORPORATION and its licensors retain all intellectual property
+# and proprietary rights in and to this software, related documentation
```
Comment (tijyojwad) [inline model_args.py]: this file is no longer needed, can be removed

File: main.py [hunk 5]
```diff
@@ -31,18 +31,20 @@
...
+from claragenomics.dl4atac import metrics
+from claragenomics.dl4atac.metrics import BCE, MSE, Recall, Specificity, CorrCoef, AUROC
+from claragenomics.dl4atac.models.models import *
+from claragenomics.dl4atac.models.model_utils import build_model
+from claragenomics.dl4atac.train import train
+from claragenomics.dl4atac.utils import *
 from claragenomics.io.bedgraphio import expand_interval, intervals_to_bg, df_to_bedGraph
 from claragenomics.io.bigwigio import bedgraph_to_bigwig
 from cmd_args import parse_args
+from model_args import model_params_args
```
Comment (tijyojwad) [inline main.py]: no longer needed

File: claragenomics/dl4atac/models/model_utils.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+_logger = logging.getLogger('AtacWorks-main')
+_handler = logging.StreamHandler()
+_handler.setLevel(logging.INFO)
+_handler.setFormatter(log_formatter)
+_logger.setLevel(logging.INFO)
+_logger.addHandler(_handler)
+
+
+def model_args_v1():
+    parser = configargparse.ArgParser(default_config_files=['configs/model_structure.yaml'])
```
Comment (tijyojwad) [inline claragenomics/dl4atac/models/model_utils.py]: let's pas sin the absolute path to the default file

File: example/run.sh [hunk 7]
```diff
@@ -109,10 +109,8 @@ echo ""
 python $root_dir/main.py --train \
     --train_files $out_dir/train_data.h5 \
     --val_files $out_dir/val_data.h5 \
-    --model resnet --nblocks 5 --nfilt 15 --width 50 \
```
Comment (tijyojwad) [inline example/run.sh]: for the run.sh file, let's explicitly specify the `model_config` option to make it clearer for users

File: claragenomics/dl4atac/train/utils.py [hunk 8]
```diff
@@ -51,6 +51,25 @@ def make_experiment_dir(label, out_home, timestamp=True, create_latest_symlink=T
         os.symlink(os.path.abspath(exp_path), latest_symlink)
     return exp_path
 
+def save_config(exp_dir, root_dir, config_file_path=None):
```
[the remark this answers is not in the document]
Comment (ntadimeti) [reply claragenomics/dl4atac/train/utils.py]: did we not discuss to leave it as it is for now ?

File: claragenomics/dl4atac/models/model_utils.py [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+import torch.multiprocessing as mp
+import torch.nn as nn
+from torch.nn.parallel import DistributedDataParallel
+from torch.optim import Adam
+
+# module imports
+from claragenomics.dl4atac.models.models import *
+from claragenomics.dl4atac.utils import *
+
+# python imports
```
[the remark this answers is not in the document]
Comment (ntadimeti) [reply claragenomics/dl4atac/models/model_utils.py]: done. removed unused imports.

File: claragenomics/dl4atac/utils.py (hunk 1 above)
Comment (ntadimeti) [reply claragenomics/dl4atac/utils.py]: copying just the model_structures.yaml for now. Will make it automatic in the next PR.

File: claragenomics/dl4atac/utils.py (hunk 2 above)
Comment (ntadimeti) [reply claragenomics/dl4atac/utils.py]: done.

File: claragenomics/dl4atac/models/model_utils.py (hunk 3 above)
Comment (ntadimeti) [reply claragenomics/dl4atac/models/model_utils.py]: done

File: model_args.py (hunk 4 above)
Comment (ntadimeti) [reply model_args.py]: done.

File: main.py (hunk 5 above)
Comment (ntadimeti) [reply main.py]: done.

File: claragenomics/dl4atac/models/model_utils.py (hunk 6 above)
Comment (ntadimeti) [reply claragenomics/dl4atac/models/model_utils.py]: done.

File: example/run.sh (hunk 7 above)
Comment (ntadimeti) [reply example/run.sh]: done.

**Pull request merged**